### PR TITLE
chore: update documentation for application controller options and application sets

### LINKED ISSUE #370: appSync option is missing from controller options documentation
**Describe the bug**
Controller options documentation needs to be updated to show use of appSync option

**To Reproduce**
Information regarding setting the appSync value for the ArgoCD controller is missing. View the documentation to see that it is not there.

**Expected behavior**
Appropriate documentation should be provided.

**Screenshots**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by tylerauerbeck**
Signed-off-by: Tyler Auerbeck <tauerbec@redhat.com>

**What type of PR is this?**

/kind documentation

**What does this PR do / why we need it**:
This is a small doc fix to add the AppSync option to the Controller documentation

**Have you updated the necessary documentation?**

* [X] Documentation update is required by this PR.
* [X] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #370 

**How to test changes / Special notes to the reviewer**:
N/A

### CODE REVIEW

File: docs/reference/argocd.md [hunk 1]
```diff
@@ -123,7 +124,8 @@ Name | Default | Description
 Processors.Operation | 10 | The number of operation processors.
 Processors.Status | 20 | The number of status processors.
 Resources | [Empty] | The container compute resources.
-LogLevel | info | The log level to be used by the ArgoCd Application Controller component. Valid options are debug, info, error, and warn.
+LogLevel | info | The log level to be used by the ArgoCD Application Controller component. Valid options are debug, info, error, and warn.
```
Comment (iam-veeramalla) [inline docs/reference/argocd.md:127]: 👍

File: docs/reference/argocd.md [hunk 2]
```diff
@@ -73,6 +73,7 @@ Name | Default | Description
 --- | --- | ---
 Image | `quay.io/argocdapplicationset/argocd-applicationset` | The container image for the ApplicationSet controller. This overrides the `ARGOCD_APPLICATIONSET_IMAGE` environment variable.
 Version | *(recent ApplicationSet version)* | The tag to use with the ApplicationSet container image.
+LogLevel | info | The log level to be used by the ArgoCD Application Controller component. Valid options are debug, info, error, and warn.
```
Comment (iam-veeramalla) [inline docs/reference/argocd.md:76]: Resources field also seem to be missing. Can you please create an entry for that as well ?

https://github.com/argoproj-labs/argocd-operator/blob/1184f4b65f628387b02b5755374e07b328a5737d/deploy/crds/argoproj.io_argocds_crd.yaml#L50

**Review by iam-veeramalla — CHANGES_REQUESTED** [review]

File: docs/reference/argocd.md (hunk 2 above)
Comment (tylerauerbeck) [reply docs/reference/argocd.md:77]: This should be resolved now.

**Comment by PixelJonas** [comment]
Just opened (and closed) another PR to add the `AppSync` property. Can I help to move this PR forward?

**Comment by tylerauerbeck** [comment]
@iam-veeramalla Anything else needed here?

**Review by iam-veeramalla — APPROVED** [review]

**Comment by iam-veeramalla** [comment]
Thanks for the PR @tylerauerbeck

**Pull request merged**